nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-security/src/test/java/org/apache/nifi/web/security/request/ContentLengthFilterTest.java

```diff
@@ -0,0 +1,183 @@
+    public void testRequestsWithMissingContentLengthHeader() throws Exception {
+        configureAndStartServer(readFullyAndRespondOK, -1);
+
+        // This shows that the ContentLengthFilter allows a request that does not have a content-length header.
+        String response = localConnector.getResponse("POST / HTTP/1.0\r\n\r\n");
+        Assert.assertFalse(StringUtils.containsIgnoreCase(response, "411 Length Required"));
+    }
+
+
+    @Test
+    public void testRequestsWithContentLengthHeader() throws Exception {
+        configureAndStartServer(readFullyAndRespondOK, -1);
+
+        int smallClaim = 150;
+        int largeClaim = 2000;
+
+        String incompletePayload = StringUtils.repeat("1", 10);
+        String largePayload = StringUtils.repeat("1", largeClaim + 200);
+
+        // This shows that the ContentLengthFilter rejects a request when the client claims more than the max + sends more than the max:
+        String response = localConnector.getResponse(String.format(POST_REQUEST, largeClaim, largePayload));
+        Assert.assertTrue(StringUtils.containsIgnoreCase(response, "413 Payload Too Large"));
+
+        // This shows that the ContentLengthFilter rejects a request when the client claims more than the max + sends less the max:
+        response = localConnector.getResponse(String.format(POST_REQUEST, largeClaim, incompletePayload));
+        Assert.assertTrue(StringUtils.containsIgnoreCase(response, "413 Payload Too Large"));
+
+        // This shows that the ContentLengthFilter allows a request when it claims less than the max + sends more than the max:
+        response = localConnector.getResponse(String.format(POST_REQUEST, smallClaim, largePayload));
+        Assert.assertTrue(StringUtils.containsIgnoreCase(response, "200 OK"));
+
+        // This shows that the server times out when the client claims less than the max + sends less than the max + sends less than it claims to send:
+        response = localConnector.getResponse(String.format(POST_REQUEST, smallClaim, incompletePayload), 500, TimeUnit.MILLISECONDS);
+        Assert.assertTrue(StringUtils.containsIgnoreCase(response, "500 Server Error"));
+        Assert.assertTrue(StringUtils.containsIgnoreCase(response, "Timeout"));
+    }
+
+
+    @Test
+    public void testJettyMaxFormSize() throws Exception {
+        // This shows that the jetty server option for 'maxFormContentSize' is insufficient for our needs because it
+        // catches requests like this:
+        configureAndStartServer(readFormAttempt, FORM_CONTENT_SIZE);
+        String form = "a=" + StringUtils.repeat("1", FORM_CONTENT_SIZE);
+        String response = localConnector.getResponse(String.format(FORM_REQUEST, form.length(), form));
+        Assert.assertTrue(StringUtils.containsIgnoreCase(response, "413 Payload Too Large"));
+
+
+        // But it does not catch requests like this:
+        response = localConnector.getResponse(String.format(POST_REQUEST, form.length(), form+form));
```

Comment:
Interesting catch. It appears it is only set for form elements. 

>The problem with maxFormContentSize it only limit the request size for "form" data and not for other content types like text/xml or text/json ... and also only if Content-Length [is sent]. So not really a strong security improvement.

From: https://stackoverflow.com/questions/26181892/configure-max-post-size-for-a-external-webapp-in-jetty#comment50923561_30646189